src/commands/reports/revoke.ts

```diff
@@ -0,0 +1,186 @@
+
+// 		let embed = new MessageEmbed()
+// 			.setTitle("FAGC Report Revocation")
+// 			.setColor("GREEN")
+// 			.setTimestamp()
+// 			.setAuthor("FAGC Community")
+// 			.setDescription(
+// 				`FAGC Report \`${report.id}\` of player \`${report.playername}\` in community ${community.name} (\`${community.id}\`)`
+// 			)
+// 		const creator = await this.client.users.fetch(report.adminId)
+// 		embed.addFields(
+// 			{ name: "Admin", value: `<@${creator.id}> | ${creator.tag}` },
+// 			{ name: "Broken rule ID", value: report.brokenRule },
+// 			{ name: "Proof", value: report.proof },
+// 			{ name: "Description", value: report.description },
+// 			{ name: "Automated", value: report.automated },
+// 			{ name: "Violated time", value: `<t:${Math.round(report.reportedTime/1000)}>` }
+// 		)
+// 		message.channel.send(embed)
+
+// 		const confirm = await getConfirmationMessage(
+// 			message,
+// 			"Are you sure you want to revoke this report?"
+// 		)
+// 		if (!confirm) return message.channel.send("Report revocation cancelled")
+
+// 		try {
+// 			const response = await this.client.fagc.reports.revoke({
+// 				reportid: reportID,
+// 				adminId: message.author.id,
+// 				reqConfig: { apikey: config.apikey }
+// 			})
+
+// 			if (response.id && response.revokedBy && response.revokedTime) {
+// 				return message.channel.send("Report revoked!")
+// 			} else {
+// 				return handleErrors(message, response)
+// 			}
+// 		} catch (error) {
+// 			if (error instanceof AuthenticationError)
+// 				return message.channel.send("Your API key is set incorrectly")
+// 			message.channel.send("Error revoking report. Please check logs.")
+// 			throw error
+// 		}
+// 	}
+// }
+// module.exports = Revoke
+
+const RevokeReport: Command = {
+	name: "revoke",
+	description: "Revokes a player's report with the report ID",
+	aliases: ["revokeid"],
+	category: "reports",
+	usage: "[reportid]",
+	examples: ["revoke p1UgG0G"],
+	requiresRoles: true,
+	requiredPermissions: ["reports"],
+	requiresApikey: true,
+	async run({ message, args, guildConfig, client }) {
+		if (!guildConfig.apiKey)
+			return message.reply(`${client.emotes.warn} No API key set`)
+		const id =
+			args.shift() ||
+			(await getMessageResponse(
+				message,
+				`${client.emotes.type} Provide a report ID to revoke`,
+			).then((x) => x?.content?.split(" ")[0]))
+		if (!id)
+			return message.channel.send(
+				`${client.emotes.warn} No report ID was provided`,
+			)
+
+		const report = await client.fagc.reports.fetchReport({ reportId: id })
+		if (!report)
+			return message.channel.send(
+				`${client.emotes.warn} Report with ID \`${id}\` doesn't exist`,
+			)
+
+		if (report.communityId !== guildConfig.communityId)
+			return message.channel.send(
+				`You are trying to revoke a report of community \`${report.communityId}\`, but you are from community \`${guildConfig.communityId}\``,
+			)
```

Comment:
To avoid infomation leak (no exist vs unauth) both should return the same message.
Suggestions: `${client.emotes.warn} Report with ID ${id} doesn't exist for this guild`

Reply:
How is fetching a report of another community an information leak?